File: VB/Scaffolding.ValidationErrors/DepartmentContextDataModel/DesignTime/DesignTimeDepartmentRepository.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.ValidationErrors.Common.Utils
Imports Scaffolding.ValidationErrors.Model
Imports Scaffolding.ValidationErrors.Common.DataModel
Imports Scaffolding.ValidationErrors.Common.DataModel.EntityFramework

Namespace Scaffolding.ValidationErrors.DepartmentContextDataModel
	Public Class DesignTimeDepartmentRepository
		Inherits DesignTimeRepository(Of Department, Integer)
		Implements IDepartmentRepository
		Public Sub New()
			MyBase.New(Function(x) x.DepartmentID)
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.ValidationErrors/DepartmentContextDataModel/DesignTime/DesignTimeDepartmentRepository.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.ValidationErrors.Common.Utils;
using Scaffolding.ValidationErrors.Model;
using Scaffolding.ValidationErrors.Common.DataModel;
using Scaffolding.ValidationErrors.Common.DataModel.EntityFramework;

namespace Scaffolding.ValidationErrors.DepartmentContextDataModel {
    public class DesignTimeDepartmentRepository : DesignTimeRepository<Department, int>, IDepartmentRepository {
        public DesignTimeDepartmentRepository()
            : base(x => x.DepartmentID) {
        }
    }
}